[candidate_parameters] Update test plan and README to include widgets (#6456)
aces/Loris: modules/candidate_parameters/test/TestPlan.md

38. Check that entering a date of death before the date of birth gives a data entry error.
39. Check that entering a date later than today gives a data entry error.
40. Check that the date stored matches the _DOD Format_ specified in the configuration module.

## Widgets
41. Access a candidate in the `candidate_profile` module

### Candidate Info Widget terms

> 42. Within the "Candidate Info" card, after the information which

Reviewer: point 42 : How is the user to test this? 
If it's not testable, it's just pre-amble for what follows, and shouldn't get a number.

    corresponds to the Candidate info returned by the LORIS API:
43. Verify that there is a "Participant Status" term with the candidate's
    status which matches the Parameter Status tab in the `candidate_parameters`
    module or "N/A" if there is no status for this candidate.
44. Verify that there is a "Caveat" term on the card with the
    description of the reason for a caveat which matches the
    Caveat Emptor flag of the "Candidate Info" tab of the
    `candidate_parameters` module, and that the term does *not*
    appear if the candidate does not have a caveat.
45. Verify that any terms from the "Candidate Parameters"
    `parameter_type_category` in step 5 of the test plan are displayed
    in the "Candidate Info" card.

### Consent Widget

> 46. With Consent configured in LORIS, verify that there is a

Reviewer: there's something unclear about "configured", and configured/enabled should be consistent across points 46-49 so i'm suggesting `enabled`. 
```suggestion
46. With Consent enabled in this LORIS, verify that there is a
```

Author: As far as I know there is no "enable/disable consent" option in LORIS. It's either configured (there are consent types set up) or it's not.

   "Consent" card in the Candidate Profile widget.

> 47. Verify that the card does *not* appear is Consent is

Reviewer: ```suggestion
47. Verify that the card does *not* appear if Consent is
```

>     not configured in LORIS.

Reviewer: ```suggestion
    not enabled in LORIS.
```

Author: See other comment.. there is no enable/disable consent in LORIS as far as I know.

Reviewer: Ok.  I don't think it's clearer now but at least it's consistent 👍

48. With Consent enabled, ensure that all types of Consent
    are displayed in a table in the card, regardless of
    whether or not the candidate being viewed has any
    data for that consent type.

> 49. Verify that rows for both "Yes" and "No" are displayed

Reviewer: fyi - "no" will be displayed first if/when this is reactified (given sort order is unspecified)

Author: There's no sorting. It's a plain table with a column for Yes/No.

    properly in the table and match the "Consent" tab of
    the module.
50. Verify that a consent type which has no data saved
    for the candidate is displayed with an empty row in

>     the table containing only a dash.

Reviewer: ```suggestion
    the table; the row should display only a dash.
```
punctuating to un-suggest the table (vs row) contains a dash
